PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/ErrorMessageTooltip.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.azure.sparkserverless.common;
+
+import com.intellij.ui.JBColor;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class ErrorMessageTooltip extends JToolTip{
+    @Nullable
+    private Popup popup;
+
+    public ErrorMessageTooltip(@NotNull JComponent component) {
+        super();
+        setComponent(component);
+        setForeground(JBColor.RED);
+    }
+
+    public void hideToolTip() {
+        if (popup != null) {
+            popup.hide();
+        }
+    }
+
+    public void showErrorToolTip(@NotNull Point location, @Nullable String errorMsg) {
+        try {
+            setTipText(errorMsg);
+            popup = PopupFactory.getSharedInstance().getPopup(getComponent(), this, location.x, location.y - 20);
+            popup.show();
+        } catch (IllegalArgumentException ignored) {
+            // This exception happens when ErrorMessageTooltip is not initialized
+        }
+    }
+
+    public void showForError(@NotNull Validatable validator) {
+        getComponent().putClientProperty("JComponent.outline", validator.isLegal() ? null : "error");
+        hideToolTip();
```

Comment:
Can be `else` of the following `if`

Reply:
Actually, I think we should change the method name `showForError` to something like `prepareForError` since we have to handle both the `validator` legal and illegal situations in this method. For both situations, we should hide previous error tooltip if it exists. That is, if we move `hideTooltip()` into `else`, error tooltips might overlap for one component.

---
In reply to: [199985755](https://github.com/Microsoft/azure-tools-for-java/pull/1747#discussion_r199985755) [](ancestors = 199985755)

Reply:
Could you add some comments here?